lib/core/versions/latest/OperationProcessor.ts

```diff
@@ -112,30 +121,25 @@ export default class OperationProcessor implements IOperationProcessor {
       return false;
     }
 
-    const operation = AnchoredOperation.createAnchoredOperation(namedAnchoredOperationModel);
+    const operation = await UpdateOperation.parse(namedAnchoredOperationModel.operationBuffer);
 
     // Verify the actual OTP hash against the expected OTP hash.
-    const isValidUpdateOtp = Multihash.isValidHash(operation.updateOtp!, didResolutionModel.metadata!.nextUpdateOtpHash!);
-    if (!isValidUpdateOtp) {
-      return false;
-    }
-
-    // The current did document must contain the public key mentioned in the operation ...
-    const publicKey = Document.getPublicKey(didDocument, operation.signingKeyId);
-    if (!publicKey) {
+    const isValidOtp = Multihash.isValidHash(operation.updateOtp, didResolutionModel.metadata!.nextUpdateOtpHash!);
+    if (!isValidOtp) {
       return false;
     }
 
-    // ... and the signature must pass verification.
-    if (!(await operation.verifySignature(publicKey))) {
+    // Verify the operation data hash against the expected operation data hash.
+    const isValidOperationData = Multihash.isValidHash(operation.encodedOperationData, operation.signedOperationDataHash.payload);
+    if (!isValidOperationData) {
       return false;
     }
 
-    // The operation passes all checks, apply the patches.
-    AnchoredOperation.applyPatchesToDidDocument(didDocument, operation.patches!);
+    const resultingDocument = await this.documentComposer.applyUpdateOperation(operation, didResolutionModel.didDocument);
 
-    didResolutionModel.metadata!.lastOperationTransactionNumber = operation.transactionNumber;
-    didResolutionModel.metadata!.nextUpdateOtpHash = operation.nextUpdateOtpHash!;
+    didResolutionModel.didDocument = resultingDocument;
+    didResolutionModel.metadata!.lastOperationTransactionNumber = namedAnchoredOperationModel.transactionNumber;
+    didResolutionModel.metadata!.nextUpdateOtpHash = operation.operationData.nextUpdateOtpHash;
```

Comment:
I'm a bit confused as to how are you able to access the `operationData.nextUpdateOtpHash` without having to import it? Also, having the same name for the internal class (in both CreateOperation.ts and UpdateOperation.ts) was going to clash right?

Reply:
Short answer is that this is something that TypeScript handles gracefully.